app/qml/ExternalResourceBundle.qml

```diff
@@ -39,17 +51,42 @@ Item {
             }
         }
 
+        /**
+         * Called when a photo is taken and confirmed (clicked on check/ok button).
+         * Original photo file is renamed with current date time to avoid name conflicts.
+         * ItemWidget reference is always set here to avoid ambiguity in case of
+         * multiple external resource (attachment) fields.
+         * \param itemWidget editorWidget for a modified field to send valueChanged signal.
+         * \param prefixToRelativePath depends on widget's config, see more qgsquickexternalwidget.qml
+         * \value depends on widget's config, see more in qgsquickexternalwidget.qml
+         */
+        property var confirmImage: function confirmImage(itemWidget, prefixToRelativePath, value) {
+          var newPath = __inputUtils.renameWithDateTime(prefixToRelativePath + "/" + value)
+          if (newPath) {
+            externalResourceHandler.itemWidget = itemWidget
+            var newCurrentValue = QgsQuick.Utils.getRelativePath(newPath, prefixToRelativePath)
+            externalResourceHandler.itemWidget.image.currentValue = newCurrentValue
+            externalResourceHandler.itemWidget.valueChanged(newCurrentValue, newCurrentValue === "" || newCurrentValue === null)
+          }
+        }
+
         property var imageSelected: function imageSelected(imagePath) {
```

Comment:
Can we have a description for imageSelected() too? I guess it's called when an image from gallery is picked and the logic is to copy it to the project directory?